build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/precommit/FormattingPrecommitPlugin.java

```diff
@@ -43,10 +41,7 @@
 
     @Override
     public void apply(Project project) {
-        final boolean shouldFormatProject = PROJECT_PATHS_TO_EXCLUDE.contains(project.getPath()) == false
-            || project.getProviders().systemProperty("es.format.everything").forUseAtConfigurationTime().isPresent();
-
-        if (shouldFormatProject) {
+        project.getPluginManager().withPlugin("java-base", javaBasePlugin -> {
```

Comment:
So will this still get applied to e.g. `build-conventions/src/*/java` ?

Reply:
It wasn't before. This plugin is defined in the `build-conventions` project, so it can't get applied to itself.